### PR TITLE
Constraint template to verify audit logging enablement

### PR DESCRIPTION
**Pull request by t12g**

### CODE REVIEW

File: policies/templates/gcp_iam_audit_log.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      validation:
+        openAPIV3Schema:
+          properties:
+            service:
+              description: "The service that should have audit logging enabled. 
+                Example: 'storage.googleapis.com' for Cloud Storage.
+                'allServices' is a special value that covers all services"
+              type: string
+            log_types:
+              description: "The log types that should be enabled. The valudes are numerical.
```
Comment (joecheuk) [inline policies/templates/gcp_iam_audit_log.yaml]: Do you think we should create the mapping underneath and let user enter the actual log type in string instead? That way is less confusing for users.

Comment (t12g) [reply policies/templates/gcp_iam_audit_log.yaml]: Done. PTAL.

**Comment by t12g** [comment]
The log types are now strings instead of ints. PTAL.

File: policies/templates/gcp_iam_audit_log.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPIAMAuditLogConstraintV1
+        plural: gcpiamauditlogconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties:
+            service:
```
Comment (joecheuk) [inline policies/templates/gcp_iam_audit_log.yaml]: if user wants to make sure DATA_READ & DATA_WRITE is enabled on say 5 apis, do we expect the user to create 5 constraints for that or should we allow them to specify that in one constraint? (i.e. having service as a list instead)

Comment (t12g) [reply policies/templates/gcp_iam_audit_log.yaml]: Done. We now accept a services list.

**Pull request merged**